components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/validators/LogoutRequestValidator.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authenticator.samlsso.logout.validators;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.opensaml.common.SAMLVersion;
+import org.opensaml.saml2.core.LogoutRequest;
+import org.wso2.carbon.identity.application.authenticator.samlsso.logout.context.SAMLMessageContext;
+import org.wso2.carbon.identity.application.authenticator.samlsso.logout.exception.SAMLIdentityException;
+import org.wso2.carbon.identity.application.authenticator.samlsso.logout.util.LambdaExceptionUtil;
+import org.wso2.carbon.identity.application.authenticator.samlsso.logout.util.SAMLFedLogoutUtil;
+import org.wso2.carbon.identity.application.authenticator.samlsso.util.SSOConstants;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static org.wso2.carbon.identity.application.authenticator.samlsso.util.SSOConstants.ISSUER_FORMAT;
+import static org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants.Authenticator.
+        SAML2SSO.IS_LOGOUT_REQ_SIGNED;
+import static org.wso2.carbon.identity.base.IdentityConstants.TRUE;
+
+/**
+ * This class is responsible for validating the SAML single logout request from the federated IdP.
+ */
+public class LogoutRequestValidator {
+
+    private static final Log log = LogFactory.getLog(LogoutRequestValidator.class);
+    private SAMLMessageContext samlMessageContext;
+
+    public LogoutRequestValidator(SAMLMessageContext samlMessageContext) {
+
+        this.samlMessageContext = samlMessageContext;
+    }
+
+    /**
+     * Validates the authentication request according to SAML SSO Web Browser Specification.
```

Comment:
This method is to validate logout requests right?

Reply:
yes

Reply:
Then change the javadoc comment accordingly. It says "Validates the authentication request ..."